apps/demo/src/app/cart/components/cart/cart.component.spec.ts

```diff
@@ -1,75 +1,196 @@
+  let cartEmptySelector: MemoizedSelector<object, boolean>;
+
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
       declarations: [
-        CartComponent,
         WrapperComponent,
-        MockCartItemComponent
+        MockCartItemsComponent,
+        MockCartSidebarComponent,
+        MockHelpBoxComponent,
+        MockCartItemCountComponent,
+        CartComponent
+      ],
+      providers: [
+        provideMockStore({})
       ]
     })
-    .compileComponents();
+      .compileComponents();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(WrapperComponent);
     wrapper = fixture.componentInstance;
+    store = TestBed.get(Store);
+    wrapper.cartValue = cart;
+    component = fixture.debugElement.query(By.css('demo-cart')).componentInstance;
 
-    wrapper.cartValue = mockCart;
+    cartItemCountSelector = store.overrideSelector(cartSelector.selectCartItemCount, stubSelectCartItemCount);
+    cartEmptySelector = store.overrideSelector(cartSelector.isCartEmpty, stubIsCartEmpty);
+
+    cartItemsElement = fixture.debugElement.query(By.css('demo-cart'));
+    cartItemCountElement = fixture.debugElement.query(By.css('demo-cart-item-count'));
+    helpBoxElement = fixture.debugElement.query(By.css('demo-help-box'));
 
     fixture.detectChanges();
-    cartItems = fixture.debugElement.queryAll(By.css('demo-cart-item'));
-    cart = fixture.debugElement.query(By.css('demo-cart')).componentInstance;
+  });
+
+  afterAll(() => {
+    store.resetSelectors();
   });
 
   it('should create', () => {
-    expect(cart).toBeTruthy();
+    expect(component).toBeTruthy();
   });
 
-  it('renders a cart', () => {
-    expect(fixture.debugElement.query(By.css('.cart'))).not.toBeNull();
+  it('should be able to take cart as input', () => {
+    expect(component.cart).toEqual(cart);
   });
 
-  it('can be passed a Cart object', () => {
-    expect(cart.cart).toEqual(mockCart);
+  describe('on <demo-cart>', () => {
+    it('should pass down the cart', () => {
+      expect(cartItemsElement.componentInstance.cart).toEqual(cart);
+    });
   });
 
-  it('renders a <demo-cart-item> for every cart.items', () => {
-    expect(cartItems.length).toEqual(mockCart.items.length);
+  describe('on <demo-cart-item-count>', () => {
+    it('should set itemCount to value grabbed from store during init', () => {
+      cartItemCountSelector.setResult(15);
+      store.setState({});
+      fixture.detectChanges();
+
+      expect(cartItemCountElement.componentInstance.itemCount).toEqual(15);
+    });
+  });
+
+  describe('when CartContainer.$loading is false', () => {
+
+    it('should render <demo-cart>', () => {
+      expect(cartItemsElement).not.toBeNull();
+    });
+
+    it('should render <demo-help-box>', () => {
+      expect(helpBoxElement).not.toBeNull();
+    });
+
+    it('should render <demo-cart-item-count>', () => {
+      expect(cartItemCountElement).not.toBeNull();
+    });
```

Comment:
lines 131-143 shouldn't be here. They were here by mistake before. They should be deleted